### PR TITLE
Feature/179 Get default nutrition by id (Repository, Service, Controller)

### PR DESCRIPTION
**Pull request by TildaEndeavour**

### CODE REVIEW

**Review by Anastasiia2023 — APPROVED** [review]

File: src/main/resources/changelog/migration.xml [hunk 1]
```diff
@@ -213,4 +213,50 @@
...
+    <changeSet id="nutritions" author="Volodymyr Bardakov" context="structure">
+
+        <comment>Create nutritions and nutrition types tables</comment>
+
+        <createTable tableName="nutrition_types">
+            <column name="id" type="bigint" autoIncrement="true">
+                <constraints primaryKey="true"/>
+            </column>
+            <column name="name" type="varchar">
+                <constraints nullable="false" unique="false"/>
```
Comment (OlegKucherenkoDev) [inline src/main/resources/changelog/migration.xml:226]: `name` field has `unique=true`

File: src/main/resources/changelog/migration.xml [hunk 2]
```diff
@@ -213,4 +213,50 @@
...
+            </column>
+            <column name="supplement" type="varchar">
+                <constraints nullable="false" unique="false"/>
+            </column>
+            <column name="recipe" type="varchar">
+                <constraints nullable="false" unique="false"/>
+            </column>
```
Comment (OlegKucherenkoDev) [inline src/main/resources/changelog/migration.xml:233]: `supplement` and `recipe` fields are redundant.

File: src/main/resources/changelog/migration.xml [hunk 3]
```diff
@@ -213,4 +213,50 @@
...
+        <createTable tableName="nutritions">
+            <column name="id" type="bigint" autoIncrement="true">
+                <constraints primaryKey="true"/>
+            </column>
+            <column name="title" type="varchar">
+                <constraints nullable="false" unique="false"/>
+            </column>
+            <column name="description" type="varchar">
+                <constraints nullable="false" unique="false"/>
+            </column>
```
Comment (OlegKucherenkoDev) [inline src/main/resources/changelog/migration.xml:245]: `description` field has `nullable=true`

File: src/main/resources/changelog/sql/default_nutritions.sql [hunk 4]
```diff
@@ -0,0 +1,25 @@
+INSERT INTO nutrition_types(id, name, supplement, recipe) VALUES
```
Comment (OlegKucherenkoDev) [inline src/main/resources/changelog/sql/default_nutritions.sql:1]:
```sql
INSERT INTO nutrition_types(id, name) VALUES (1, 'Recipe'), (2, 'Supplement');
```

File: src/main/resources/changelog/sql/default_nutritions.sql [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+(3, 'Energizing Fruit Smoothie', 'A refreshing blend of mixed berries, banana, almond milk, and a scoop of protein powder.', false, 3),
+(4, 'Whole Grain Pasta Salad', 'A hearty salad featuring whole grain pasta, cherry tomatoes, olives, and feta cheese with a lemon-herb dressing.', false, 4),
+(5, 'Lean Turkey Wrap', 'A light and satisfying wrap filled with sliced turkey, lettuce, tomato, and a hint of mustard.', false, 5),
+(6, 'Omega-3 Rich Salmon', 'Grilled salmon fillet served with steamed asparagus and a side of quinoa.', false, 6),
+(7, 'Fiber-Rich Bean Soup', 'A comforting soup made with mixed beans, vegetables, and herbs, providing a good dose of fiber.', false, 7),
+(8, 'Low-Carb Cauliflower Rice Stir-Fry', 'A delicious stir-fry dish using cauliflower rice, mixed vegetables, and lean chicken strips.', false, 8),
+(9, 'Mediterranean Chickpea Salad', 'A vibrant salad packed with chickpeas, cucumbers, red onions, feta cheese, and a lemon dressing.', false, 9),
+(10, 'High-Protein Egg Breakfast', 'A satisfying breakfast featuring scrambled eggs, spinach, tomatoes, and whole grain toast.', false, 10);
```
Comment (OlegKucherenkoDev) [inline src/main/resources/changelog/sql/default_nutritions.sql:25]: These all are Recipes, so `nutrition_type_id` will be 1.
Examples of Supplements are: Multivitamins and Minerals, Omega, Protein, Q10, Lecithin, Collagen. They will have `nutrition_type_id` equaled to 2:

```sql
(11, 'Multivitamins and Minerals', 'Elevate your overall health with essential vitamins and minerals for a well-rounded nutritional foundation.', false, 2),
(12, 'Protein Shake', 'Fuel your body with the daily protein it needs for muscle repair, growth, and overall strength.', false, 2),
(13, 'Q10', 'Revitalize your energy levels and fortify your immune system with Q10, a powerful antioxidant supporting cellular energy production.', false, 2),
(14, 'Omega', 'Nourish your body with essential fatty acids for heart health, cognitive function, and overall well-being.', false, 2),
(15, 'Lecithin', 'Unlock the benefits of choline, an essential nutrient found in lecithin, supporting brain function and overall cognitive well-being.', false, 2),
(16, 'Collagen', 'Nurture your skin, hair, nails, and bones with collagen, promoting elasticity, strength, and structural integrity for a vibrant appearance and overall well-being.', false, 2);

ALTER sequence nutritions_id_seq restart with 17;
```

**Review by OlegKucherenkoDev — CHANGES_REQUESTED** [review]

**Review by OlegKucherenkoDev — CHANGES_REQUESTED** [review]

**Review by OlegKucherenkoDev — APPROVED** [review]

**Pull request merged**